client/ayon_core/pipeline/workfile/path_resolving.py

```diff
@@ -278,42 +273,141 @@ def get_last_workfile_with_version(
+    output_filepaths = []
+    for filepath in sorted(filepaths):
+        filename = os.path.basename(filepath)
         match = re.match(file_template, filename, **kwargs)
         if not match:
             continue
 
         if not match.groups():
-            output_filenames.append(filename)
+            output_filepaths.append(filename)
             continue
 
         file_version = int(match.group(1))
         if version is None or file_version > version:
-            output_filenames[:] = []
+            output_filepaths.clear()
             version = file_version
 
         if file_version == version:
-            output_filenames.append(filename)
+            output_filepaths.append(filepath)
+
+    output_filepath = None
+    last_time = None
+    for _output_filepath in output_filepaths:
+        mod_time = None
+        if os.path.exists(_output_filepath):
+            mod_time = os.path.getmtime(_output_filepath)
+        if (
+            last_time is None
+            or (mod_time is not None and last_time < mod_time)
+        ):
+            output_filepath = _output_filepath
+            last_time = mod_time
+
+    return output_filepath, version
+
+
+def get_last_workfile_from_paths(
+    filepaths: list[str],
+    file_template: str,
+    template_data: dict[str, Any],
+    extensions: set[str],
+):
+    """Return last workfile filename.
+
+    Returns file with version 1 if there is not workfile yet.
+
+    Args:
+        filepaths (list[str]): Paths to workfiles.
+        file_template (str): Template of file name.
+        template_data (dict[str, Any]): Data for filling template.
+        extensions (set[str]): All allowed file extensions of workfile.
+
+    Returns:
+        Optional[str]: Last or first workfile as filename of full path
+            to filename.
+
+    """
+    filepath, _version = get_last_workfile_with_version_from_paths(
+        filepaths, file_template, template_data, extensions
+    )
+    return filepath
+
+
+def _filter_dir_files_by_ext(
+    dirpath: str,
+    extensions: set[str],
+):
+    """Filter files by extensions.
+
+    Args:
+        dirpath (str): List of file paths.
+        extensions (set[str]): Set of file extensions.
+
+    Returns:
+        tuple[list[str], set[str]]: Filtered list of file paths.
+
+    """
+    dotted_extensions = set()
+    for ext in extensions:
+        if not ext.startswith("."):
+            ext = f".{ext}"
+        dotted_extensions.add(ext)
+    filtered_paths = [
+        os.path.join(dirpath, filename)
+        for filename in os.listdir(dirpath)
+        if os.path.splitext(filename)[-1] in dotted_extensions
+    ]
+    return filtered_paths, dotted_extensions
+
+
+def get_last_workfile_with_version(
+    workdir, file_template, fill_data, extensions
+):
+    """Return last workfile version.
+
+    Usign workfile template and it's filling data find most possible last
+    version of workfile which was created for the context.
```

Comment:
We already changed in this PR `fill_data` to `template_data`, shouldn't we do the same here?

```suggestion
def get_last_workfile_with_version(
    workdir: str, file_template: str, fill_data: dict[str, Any], extensions: Iterable[str]
) -> tuple[Optional[str], Optional[int]]:
    """Return the last workfile version.

    Using the workfile template and its filling data to find the most possible last
    version of the workfile which was created for the context.
```